File: ALGO/Python/Sort/selection_sort.py
from random import randint


def selection_sort(a):
    """
    Selection Sort algorithm function
    @param a : list to sort
    """
    for i in enumerate(a):
        min_index = a[i[0]:].index(min(a[i[0]:]))
        min_index += i[0]
        a[i[0]], a[min_index] = a[min_index], a[i[0]]
    return a


if __name__ == '__main__':
    n = int(input("Enter the number of elements you want: "))
    ARR = [randint(0, 1000) for _ in range(n)]
    ARR = selection_sort(ARR)
    print(ARR)

File: ALGO/C/Sort/selectionSort.c
#include <stdio.h>
#include <stdlib.h>

selectionSort(int *arr, int n)
{
    int i, j, min, temp;
    for (i = 0; i < n; i++)
    {
        min = i;
        for (j = i + 1; j < n; j++)
            if (arr[j] < arr[min])
                min = j;
        if (min != i)
        {
            temp = arr[min];
            arr[min] = arr[i];
            arr[i] = temp;
        }
    }
}

int main(int argc, char const *argv[])
{
    int i;
    int arr[] = {56, 23, 45, 54, 76, 98, 34, 54, 12, 1};
    selectionSort(arr, 10);
    for (i = 0; i < 10; i++)
        printf("%d ", arr[i]);
    return 0;
}
